src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/DesignTimeCSharpRenderer.cs

```diff
@@ -0,0 +1,355 @@
+            }
+
+            // Wrap the directive token in a lambda to isolate variable names.
+            Context.Writer
+                .Write("((")
+                .Write(typeof(Action).FullName)
+                .Write(")(");
+            using (Context.Writer.BuildLambda(endLine: false))
+            {
+                var originalIndent = Context.Writer.CurrentIndent;
+                Context.Writer.ResetIndent();
+                switch (tokenKind)
+                {
+                    case DirectiveTokenKind.Type:
+
+                        AddLineMappingFor(node);
+                        Context.Writer
+                            .Write(node.Content)
+                            .Write(" ")
+                            .WriteStartAssignment(TypeHelper)
+                            .WriteLine("null;");
+                        break;
+                    case DirectiveTokenKind.Member:
+                        Context.Writer
+                            .Write(typeof(object).FullName)
+                            .Write(" ");
+
+                        AddLineMappingFor(node);
+                        Context.Writer
+                            .Write(node.Content)
+                            .WriteLine(" = null;");
+                        break;
+                    case DirectiveTokenKind.String:
+                        Context.Writer
+                            .Write(typeof(object).FullName)
+                            .Write(" ")
+                            .WriteStartAssignment(TypeHelper);
+
+                        if (node.Content.StartsWith("\"", StringComparison.Ordinal))
+                        {
+                            AddLineMappingFor(node);
+                            Context.Writer.Write(node.Content);
+                        }
+                        else
+                        {
+                            Context.Writer.Write("\"");
+                            AddLineMappingFor(node);
+                            Context.Writer
+                                .Write(node.Content)
+                                .Write("\"");
+                        }
+
+                        Context.Writer.WriteLine(";");
+                        break;
+                }
+                Context.Writer.SetIndent(originalIndent);
+            }
+            Context.Writer.WriteLine("))();");
+
+        }
+
+        public override void VisitTemplate(TemplateIRNode node)
+        {
+            const string ItemParameterName = "item";
+            const string TemplateWriterName = "__razor_template_writer";
+
+            Context.Writer
+                .Write(ItemParameterName).Write(" => ")
+                .WriteStartNewObject("Microsoft.AspNetCore.Mvc.Razor.HelperResult" /* ORIGINAL: TemplateTypeName */);
+
+            var initialRenderingConventions = Context.RenderingConventions;
+            var redirectConventions = new CSharpRedirectRenderingConventions(TemplateWriterName, Context.Writer);
```

Comment:
Shouldn't need these bits at design time since all write calls result in `__o = ...`